добавление все требуемых функция в рамках группового проекта

tukanaft opened the pull request.

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

tukanaft committed:
init

TimmyJimmy5 committed:
Первая итерация режиссеров.

TimmyJimmy5 committed:
Вторая итерация режиссеров.

tukanaft committed:
init

tukanaft committed:
style fix

tukanaft committed:
postman fix

tukanaft committed:
postman fix

tukanaft committed:
Merge pull request #9 from tukanaft/add-director

Первая итерация режиссеров.

tukanaft committed:
Merge branch 'develop' into add-most-populars

tukanaft committed:
Merge pull request #11 from tukanaft/add-most-populars

add most mopular films by genre and year

tukanaft committed:
Update FilmRepository.java

conflict resolve

tukanaft committed:
Update FilmRepository.java

resole conflict

tukanaft committed:
merge conflicts fix

AndreyDidan committed:
Добавил логику вывода общих с другом фильмов с сортировкой по их популярности.

AndreyDidan committed:
Реализовал простую рекомендательную систему для фильмов.

AndreyDidan committed:
Поправил чекстайл

AndreyDidan committed:
Опять поправил чекстайл

AndreyDidan committed:
Опять поправил чекстайл

tukanaft committed:
merge conflicts fix

tukanaft committed:
Merge pull request #15 from tukanaft/add-recommendations

Реализовал простую рекомендательную систему для фильмов.

tukanaft committed:
Merge branch 'develop' into add-common-films

tukanaft committed:
Merge pull request #14 from tukanaft/add-common-films

Добавил логику вывода общих с другом фильмов с сортировкой по их попу…

TimmyJimmy5 committed:
Реализован поиск по директору и по названию фильма

TimmyJimmy5 committed:
Правка опечатки

LudoEd112 committed:
коммит 1, Выполнено задание на удаление пользователя и фильма

LudoEd112 committed:
коммит 1, Выполнено задание на удаление пользователя и фильма

TimmyJimmy5 committed:
Правка очередности вызова методов поиска в котроллере для поиска по директору и по фильму

tukanaft committed:
Merge pull request #16 from tukanaft/add-search

Реализован поиск по директору и по названию фильма

tukanaft committed:
Merge pull request #17 from tukanaft/add-remove-endpoint

коммит 1, Выполнено задание на удаление пользователя и фильма

togean committed:
Добавлен функционал отзывов. Добавление, редактирование, удаление, получение по ID, вывод по фильмам, установка и удаление пользователем лайков отзывам

togean committed:
Merge remote-tracking branch 'origin/develop' into add-reviews

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/dao/mappers/UserRowMapper.java
#	src/main/resources/schema.sql

togean committed:
Добавлен функционал отзывов. Добавление, редактирование, удаление, получение по ID, вывод по фильмам, установка и удаление пользователем лайков отзывам. Пока 2 теста до концп не проходят - ещё доработаю.

togean committed:
Добавлен функционал отзывов. Добавление, редактирование, удаление, получение по ID, вывод по фильмам, установка и удаление пользователем лайков отзывам. Пока 2 теста до конца не проходят - ещё доработаю.

togean committed:
Переделал под новую структуру проекта с которой коллеги работали, подправил логику для прохода тестов

tukanaft committed:
Merge pull request #19 from tukanaft/add-reviews

Добавлен функционал работы от отзывами

tukanaft committed:
фикс расположения методов

LudoEd112 committed:
коммит 1, выполнены основные функции, но пройдены не все тесты

LudoEd112 committed:
коммит 2, выполнены основные функции, но пройдены не все тесты, почти закончено

LudoEd112 committed:
коммит 3, все тесты пройдены

tukanaft committed:
Merge pull request #20 from tukanaft/add-feed

Add feed

tukanaft committed:
postman test fix

togean committed:
Поправил работу под общие тесты. Ошибок по блоку отзывов нет.

togean committed:
Поправил работу под общие тесты. Ошибок по блоку отзывов нет.

togean committed:
Поправил работу под общие тесты. Ошибок по блоку отзывов нет.

tukanaft committed:
Merge pull request #22 from tukanaft/add-reviews

Обновление reviews  для работы с общими тестами

LudoEd112 committed:
коммит 4, не все тесты пройдены

LudoEd112 committed:
коммит 5, все тесты пройдены

tukanaft committed:
Merge pull request #23 from tukanaft/add-feed

Add feed

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/EventController.java:
```diff
@@ -0,0 +1,28 @@
+    @GetMapping
+    public List<Event> getUserFeed(@PathVariable Long id) {
+        List<Event> event = eventService.getUserFeed(id);
+        System.out.println(event.toArray().length);
```
думаю это стоит сделать через логер, тем более он без дела стоит))

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,69 +1,114 @@
 
+    @GetMapping("/search")
+    public List<FilmDto> searchFilms(@RequestParam("query") String query, @RequestParam("by") String by) {
+        switch (by) {
```
1) всю эту логику нужно перенести в filmService в соответствующий метод 
2) лучше спарсить атрибут by по признаку запятая и собрать собрать список, гораздо проще будет проверять по какому признаку искать фильмы

TimmyJimmy5 replied:
1. Тут согласен. Перенес логику в сервис.
2. Оптимизировал switch. Больше править не буду, т.к. уже не успею по причине того, что в ближайшие дни другие дела есть, а дедлайн 4.11 + то, что сейчас, читабельно и работает хорошо, а как известно: работает - не трогай)
Да и честно говоря не вижу других способов сделать это еще проще.

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,69 +1,114 @@
+                return searchDirector.stream().distinct().collect(Collectors.toList());
+            }
+            default -> {
+                return null;
```
если в by пришло что-то отличное от title и director, то нужно ответить исключением

TimmyJimmy5 replied:
Согласен, сделано.

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/ReviewController.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
```
в этом классе нет логирования

togean replied:
Добавил полезное логирование

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,66 +1,83 @@
 
-@Slf4j
 @RestController
+@Slf4j
```
и в этом тоже

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/FilmRepository.java:
```diff
@@ -0,0 +1,211 @@
+    }
+
+    public boolean delete(Long id) {
+        return delete(DELETE_QUERY, id);
```
фильм нужно удалять и из смежных таблиц типа film_genres, film_user_likes_set, film_directors

tukanaft replied:
оно удаляется по смотрите файл схемы

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/ReviewRepository.java:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    public Review addReview(Review newReview) {
+        newReview.setUseful(0);
```
это уже сделано в классе InMemoryReviewService

togean replied:
Да, точно. Убрал лишнее

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/ReviewRepository.java:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    public Optional<Review> updateReviewWithoutUseful(Review newReview) {
+        if (isReviewExists(newReview.getReviewId())) {
```
цепочке обновления ревью это уже третья проверка на существование ревью. Такое частое обращение к БД сильно тормозит приложение

togean replied:
Да, тут избыточно. Убрал проверку в этом методе

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/ReviewRepository.java:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    public void deleteReview(Long reviewId) {
+        if (isReviewExists(reviewId)) {
```
в цепочке удаления ревью это уже третья проверка на существование ревью. Такое частое обращение к БД сильно тормозит приложение

togean replied:
Да, тут тоже убрал лишнее

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/ReviewRepository.java:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    public Boolean isReviewExists(Long userId) {
+        int count = jdbc.queryForObject(CHECK_REVIEW_EXISTS_QUERY, Integer.class, userId);
```
count нужно сделать Integer, поскольку данный запрос может вернуть null, а в примитив null не запишешь. Соответственно в ответ добавится return count != null && count > 0

togean replied:
Согласен. Исправил

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/dao/repository/UserRepository.java:
```diff
@@ -0,0 +1,143 @@
+    }
+
+    public boolean delete(Long id) {
+        return delete(DELETE_QUERY, id);
```
пользователя нужно удалять еще из смежных таблиц film_user_likes_set, user_friends

tukanaft replied:
оно удаляется по смотрите файл схемы

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/InMemoryReviewService.java:
```diff
@@ -0,0 +1,125 @@
+
+    @Override
+    public Optional<Review> updateReview(Review newReview) {
+        if (!reviewRepository.isReviewExists(newReview.getReviewId())) {
```
нет необходимости специально проверять существование ревью, поскольку в методе reviewRepository.getReviewById() предварительно делается тоже самое

togean replied:
Да, исправил

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/InMemoryReviewService.java:
```diff
@@ -0,0 +1,125 @@
+        if (!reviewRepository.isReviewExists(reviewId)) {
+            throw new NotFoundException("Такого отзыва не существует");
+        }
+        if (inMemoryUserService.get(userId) == null) {
```
метод inMemoryUserService.get() сам бросает исключение, если пользователь не найден, можно просто вызвать метод

togean replied:
Исправил

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/InMemoryReviewService.java:
```diff
@@ -0,0 +1,125 @@
+        if (newReview.getIsPositive() == null) {
+            throw new BadInputException("Не указан ID фильма");
+        }
+        if (inMemoryUserService.get(newReview.getUserId()) == null) {
```
метод inMemoryUserService.get() сам бросает исключение, если пользователь не найден, можно просто вызвать метод

togean replied:
Согласен, исправил

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/InMemoryReviewService.java:
```diff
@@ -0,0 +1,125 @@
+        if (inMemoryUserService.get(newReview.getUserId()) == null) {
+            throw new NotFoundException("Указанного пользователя для отзыва не существует");
+        }
+        if (inMemoryFilmService.findFilmById(newReview.getFilmId()) == null) {
```
аналогично

togean replied:
Тоже исправил

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/InMemoryReviewService.java:
```diff
@@ -0,0 +1,125 @@
+
+    @Override
+    public void deleteReview(Long reviewId) {
+        if (!reviewRepository.isReviewExists(reviewId)) {
```
нет необходимости специально проверять существование ревью, поскольку в методе reviewRepository.getReviewById() предварительно делается тоже самое

LudoEd112 committed:
коммит 6, исправлены недочеты

tukanaft committed:
Merge pull request #24 from tukanaft/add-feed

коммит 6, исправлены недочеты

TimmyJimmy5 committed:
Перенесена логика поиска в сервис.
Добавлен выброс исключений, если будет получено что-то отличное от директора или тайтла в by

tukanaft committed:
Merge pull request #25 from tukanaft/fix-search

Перенесена логика поиска в сервис.

TimmyJimmy5 committed:
Переделан case

tukanaft committed:
Merge pull request #26 from tukanaft/fix-search

Переделан case

togean committed:
Исправил код по комментариям ревьюера

togean committed:
Исправил код по комментариям ревьюера

togean committed:
Исправил код по комментариям ревьюера, подправил стили

togean committed:
Нашёл ещё пару мест с подобной ошибкой, как указывал ревьюер, но он их не указал. Я, тем не менее, исправил. Тесты все прошли.

togean committed:
Нашёл ещё пару мест с подобной ошибкой, как указывал ревьюер, но он их не указал. Я, тем не менее, исправил. Тесты все прошли.

tukanaft committed:
Merge pull request #27 from tukanaft/add-reviews

Отработка кода по комментариям ревьюера
